leetcode/src/69.c

```diff
@@ -0,0 +1,24 @@
+//using binary search method is one of the efficient one for this problem statement.
+int mySqrt(int x){
+int start=0;
+        int end=x;
+        long long int ans=0;
+        while(start <= end){
+               long long int mid=(start+end)/2;
+                long long int val=mid*mid;
+               if(mid*mid == x){
+                         return mid;
+                }
+//if mid is less than the square root of the number(x) store the value of mid in ans.
+               if( mid*mid < x){
```

Comment:
You have the `val=mid*mid` but you still multiply it 3 times and check the value. You can use val instead.